Dev binding tree

FantasmasDev opened the pull request.

FantasmasDev committed:
1. Добавлены кнопки и TextView

FantasmasDev committed:
1. Добавлены 2 экрана

FantasmasDev committed:
1. созданы экраны: share, libray
2. убрана заглушка toast
3. реализован переход между экранами

FantasmasDev committed:
переработаны файлы ресурсов
1. добавлены размеры.
2. изменён метод возврата на основной экран
3. доработана подстановка цвета из ресурсов
4. подстановка размеров из ресурсов

FantasmasDev committed:
1. переписан на binding

FantasmasDev committed:
1. переделан под использование стилей
2. добавлены шрифты

FantasmasDev committed:
1. инициализирована ночная тема

FantasmasDev committed:
1. исправлена тема

FantasmasDev committed:
1. добавлены действия для кнопки "поделиться приложением"
2. добавлены действия для кнопки "поддержки"
3. добавлены действия для кнопки "пользовательское соглашение"

FantasmasDev committed:
1. инициализирован экран поиска

FantasmasDev committed:
1. доделан экран поиска, добавлена тема

FantasmasDev committed:
1. переделан search bar на группу элементов
2. реализована часть логики экрана поиска

FantasmasDev committed:
1. кнопка сброс теперь, скрывает клавиатуру

FantasmasDev committed:
1. временная реализация хранения данных

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,30 @@
+        setContentView(binding.root)
+
+        binding.searchButton.setOnClickListener {
+            val searchIntent = Intent(this@MainActivity, SearchActivity::class.java)
```
Зачем тут this@MainActivity, если в других аналогичных кнопках нет и оно тоже работает?

FantasmasDev replied:
убрал

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,72 @@
+        binding = ActivitySearchBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+//        if (savedInstanceState != null) {
```
Законментированный код мы не пушим

FantasmasDev replied:
убрал комменты, оставил в стилях, мне они нужны на будущее

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,72 @@
+        super.onRestoreInstanceState(savedInstanceState)
+        // binding.searchBar.setText()
+        val text = savedInstanceState.getString("USER_INPUT")
+        Toast.makeText(this@SearchActivity, text + "2", Toast.LENGTH_SHORT).show()
```
В требованиях нигде не сказано, что должны быть тосты

FantasmasDev replied:
тосты добавлял для проверок самого себя, сейчас убраны

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+            shareIntent.type = "text/plain"
+            shareIntent.putExtra(
+                Intent.EXTRA_TEXT,
+                "https://practicum.yandex.ru/android-developer/?from=catalog"
```
Эту строчку в ресурсы нужно вынести

FantasmasDev replied:
вынес

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+                Intent.EXTRA_TEXT,
+                "https://practicum.yandex.ru/android-developer/?from=catalog"
+            )
+            startActivity(Intent.createChooser(shareIntent, "Поделиться приложением"))
```
И тут

FantasmasDev replied:
и тут тоже исправил

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+        binding.supportButton.setOnClickListener {
+            val supportIntent = Intent(Intent.ACTION_SENDTO)
+            supportIntent.data = Uri.parse("mailto:")
+            supportIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("fragofikus@gmail.com"))
```
И эту

FantasmasDev replied:
и тут исправил

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+            supportIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("fragofikus@gmail.com"))
+            supportIntent.putExtra(
+                Intent.EXTRA_SUBJECT,
+                "Сообщение разработчикам и разработчицам приложения Playlist Maker"
```
И тут

FantasmasDev replied:
и эту исправил

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+            )
+            supportIntent.putExtra(
+                Intent.EXTRA_TEXT,
+                "Спасибо разработчикам и разработчицам за крутое приложение!"
```
И тут

FantasmasDev replied:
исправил

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,51 @@
+
+        binding.termsOfUseButton.setOnClickListener {
+            val termsIntent =
+                Intent(Intent.ACTION_VIEW, Uri.parse("https://yandex.ru/legal/practicum_offer/"))
```
И вот эту

FantasmasDev replied:
исправил

AleksandrIlinskii commented on app/src/main/res/values/colors.xml:
```diff
@@ -2,4 +2,9 @@
     <color name="white">#FFFFFFFF</color>
+    <color name="main_background_blue">#3772E7</color>
+    <color name="main_background_black">#1A1B22</color>
+    <color name="main_font_black">#1A1B22</color>
```
main_background_black и main_font_black - чем отличаются эти цвета?

FantasmasDev replied:
фон и шрифт, чтобы при смене одного другой не слетал

AleksandrIlinskii commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,15 @@
-    <string name="app_name">Playlist Maker</string>
+    <string name="app_name" translatable="false">Playlist Maker</string>
+
+    <string name="main_header" translatable="false">Playlist Maker</string>
```
Зачем эта строчка, если можно просто взять app_name?

FantasmasDev replied:
На случай если шапка разойдётся с названием приложения

FantasmasDev committed:
1. внесены правки по ревью

FantasmasDev committed:
1. чистка кода

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -18,7 +18,26 @@
+    <!--            android:src="@drawable/clear_button_cross"-->
+    <!--            android:visibility="gone"-->
+    <!--            app:tint="?attr/colorAccent" />-->
+    <!--    </LinearLayout>-->
```
Это все можно удалить

FantasmasDev committed:
1. Реализован ресайклер вью
2. Реализован класс Track
3. Реализован класс Adapter
4. Реализован класс Holder
5. Реализована прокрутка списка
6. Доработан стиль кода
7. Добавлен Glide

FantasmasDev committed:
1. Реализован Glide для загрузки картинок
